node.hpp

```diff
@@ -0,0 +1,198 @@
+#include <stdlib.h>
+#include <stdio.h>
+
+using namespace std;
+
+class Node
+{
+    public:
+        Node();
+        ~Node();
+
+        int get_value();
+        void set_value(int value);
+        int get_key();
```

Comment:
you may want to look into usng safe data types in your code. int can have different implementations on different platforms, etc

http://www.codesynthesis.com/~boris/blog/2008/10/13/writing-64-bit-safe-code/